deepchecks/nlp/checks/data_integrity/text_property_outliers.py

```diff
@@ -73,85 +73,102 @@ def run_logic(self, context: Context, dataset_kind: DatasetKind) -> CheckResult:
-                # For the upper and lower limits doesn't show values that are smaller/larger than the actual values
-                # we have in the data
-                'lower_limit': max(lower_limit, min(values_arr)),
-                'upper_limit': min(upper_limit, max(values_arr)) if is_numeric else None,
-                'outlier_ratio': len(text_outliers) / len(values_arr)
-            }
+                if is_numeric:
+                    lower_limit, upper_limit = iqr_outliers_range(values_arr, self.iqr_percentiles,
+                                                                  self.iqr_scale, self.sharp_drop_ratio)
+                else:
+                    # Counting the frequency of each category. Normalizing because distribution graph shows percentage.
+                    counts_map = pd.Series(values_arr.astype(str)).value_counts(normalize=True).to_dict()
+                    lower_limit = sharp_drop_outliers_range(sorted(list(counts_map.values()), reverse=True),
+                                                            self.sharp_drop_ratio) or 0
+                    upper_limit = len(values_arr)  # No upper limit for categorical properties
+                    values_arr = np.array([counts_map[x] for x in values_arr])  # replace the values with the counts
+
+                # Get the indices of the outliers
+                top_outliers = np.argwhere(values_arr > upper_limit).squeeze(axis=1)
+                # Sort the indices of the outliers by the original values
+                top_outliers = top_outliers[
+                    np.apply_along_axis(lambda i, sort_arr=values_arr: sort_arr[i], axis=0, arr=top_outliers).argsort()
+                ]
+
+                # Doing the same for bottom outliers
+                bottom_outliers = np.argwhere(values_arr < lower_limit).squeeze(axis=1)
+                # Sort the indices of the outliers by the original values
+                bottom_outliers = bottom_outliers[
+                    np.apply_along_axis(lambda i, sort_arr=values_arr: sort_arr[i],
+                                        axis=0, arr=bottom_outliers).argsort()
+                ]
+
+                text_outliers = np.concatenate([bottom_outliers, top_outliers])
+
+                result[name] = {
+                    'indices': [dataset.get_original_text_indexes()[i] for i in text_outliers],
+                    # For the upper and lower limits doesn't show values that are smaller/larger than the actual values
+                    # we have in the data
+                    'lower_limit': max(lower_limit, min(values_arr)),
+                    'upper_limit': min(upper_limit, max(values_arr)) if is_numeric else None,
+                    'outlier_ratio': len(text_outliers) / len(values_arr),
+                }
+            except Exception as exp:  # pylint: disable=broad-except
+                result[name] = f'{exp}'
```

Comment:
make sure exp is printed out nicely

Reply:
Verified and exp will just be an error message and resolving this comment.